test/models/podcast_planner_test.rb

```diff
@@ -0,0 +1,423 @@
+          planner.period = 2
+          planner.generate_dates!
+          assert_equal planner.dates, [
+            # Monday, March 5
+            DateTime.new(2001, 3, 5),
+            # Monday, March 19
+            DateTime.new(2001, 3, 19),
+            # Monday, April 2
+            DateTime.new(2001, 4, 2),
+            # Monday, April 16
+            DateTime.new(2001, 4, 16),
+            # Monday, April 30
+            DateTime.new(2001, 4, 30),
+            # Monday, May 14
+            DateTime.new(2001, 5, 14),
+            # Monday, May 28
+            DateTime.new(2001, 5, 28),
+            # Monday, June 11
+            DateTime.new(2001, 6, 11),
+            # Monday, June 25
+            DateTime.new(2001, 6, 25),
+            # Monday, July 9
+            DateTime.new(2001, 7, 9),
+            # Monday, July 23
+            DateTime.new(2001, 7, 23),
+            # Monday, August 6
+            DateTime.new(2001, 8, 6),
+            # Monday, August 20
+            DateTime.new(2001, 8, 20)
+          ]
+          planner.dates.each do |date|
+            assert_equal date > planner.start_date, true
+            assert_equal date <= planner.end_date, true
+          end
+
+          # every 4 weeks
+          planner.period = 4
+          planner.generate_dates!
+          assert_equal planner.dates, [
+            # Monday, March 5
+            DateTime.new(2001, 3, 5),
+            # Monday, April 2
+            DateTime.new(2001, 4, 2),
+            # Monday, April 30
+            DateTime.new(2001, 4, 30),
+            # Monday, May 28
+            DateTime.new(2001, 5, 28),
+            # Monday, June 25
+            DateTime.new(2001, 6, 25),
+            # Monday, July 23
+            DateTime.new(2001, 7, 23),
+            # Monday, August 20
+            DateTime.new(2001, 8, 20)
+          ]
+          planner.dates.each do |date|
+            assert_equal date > planner.start_date, true
+            assert_equal date <= planner.end_date, true
+          end
+        end
+      end
+    end
+
+    describe "generate_drafts!" do
+      before do
+        planner.dates = [
+          # Monday, March 5
+          DateTime.new(2001, 3, 5),
+          # Wednesday, March 7
+          DateTime.new(2001, 3, 7),
+          # Monday, April 2
+          DateTime.new(2001, 4, 2),
+          # Wednesday, April 4
+          DateTime.new(2001, 4, 4),
+          # Wednesday, May 2
+          DateTime.new(2001, 5, 2),
+          # Monday, May 7
+          DateTime.new(2001, 5, 7),
+          # Monday, June 4
+          DateTime.new(2001, 6, 4),
+          # Wednesday, June 6
+          DateTime.new(2001, 6, 6),
+          # Monday, July 2
+          DateTime.new(2001, 7, 2),
+          # Wednesday, July 4
+          DateTime.new(2001, 7, 4)
+        ]
+        # 9:30 AM
+        planner.publish_time = DateTime.new(2001, 2, 3, 9, 30)
+        planner.segment_count = 2
+      end
+
+      describe "Episodes" do
```

Comment:
i actually wasn't sure what i needed to test in this suite, so i just check that the `Episode`s look correct according to the `dates`, `publish_time`, and `segment_count` supplied.

Reply:
Makes sense.  You could also have skipped `.save!`-ing these episodes, and just look at the `planner.drafts` array (would be slightly faster to not hit the db ... just assert that they look `.valid?`).